python/selfie-lib/selfie_lib/PerCharacterEscaper.py

```diff
@@ -0,0 +1,92 @@
+class PerCharacterEscaper:
+    def __init__(self, escape_code_point, escaped_code_points, escaped_by_code_points):
+        self.escape_code_point = escape_code_point
+        self.escaped_code_points = escaped_code_points
+        self.escaped_by_code_points = escaped_by_code_points
+
+    @staticmethod
+    def _first_offset_needing_escape(
```

Comment:
- private methods have two underscores, not one
- why is this static? it seems error-prone to pass `escaped_code_points` without the self